ompi/mpi/c/intercomm_create.c

```diff
@@ -116,16 +131,31 @@ int MPI_Intercomm_create(MPI_Comm local_comm, int local_leader,
         rc = MCA_PML_CALL(irecv(&rsize, 1, MPI_INT, rleader, tag, bridge_comm,
                                 &req));
         if ( rc != MPI_SUCCESS ) {
+#if OPAL_ENABLE_FT_MPI
+            if( MPI_ERR_PROC_FAILED == rc ) {
+                rsize = 0;
+                goto skip_handshake;
+            }
+#endif  /* OPAL_ENABLE_FT_MPI */
             goto err_exit;
         }
         rc = MCA_PML_CALL(send (&local_size, 1, MPI_INT, rleader, tag,
                                 MCA_PML_BASE_SEND_STANDARD, bridge_comm));
         if ( rc != MPI_SUCCESS ) {
+#if OPAL_ENABLE_FT_MPI
+            if( MPI_ERR_PROC_FAILED == rc ) {
+                rsize = 0;
+                goto skip_handshake;
+            }
+#endif  /* OPAL_ENABLE_FT_MPI */
             goto err_exit;
         }
+#if OPAL_ENABLE_FT_MPI
+  skip_handshake:  /* nothing special */;
+#endif /* OPAL_ENABLE_FT_MPI */
         rc = ompi_request_wait( &req, MPI_STATUS_IGNORE);
         if ( rc != MPI_SUCCESS ) {
-            goto err_exit;
+            rsize = 0;  /* participate in the collective and then done */
```

Comment:
This changes the behavior of the code even if FT_MPI is not ON (but only in the case of error, so maybe this is not a problem, just pointing out).

Reply:
Yes, the code would deadlock just as good with  non-FT related errors.